src/core/components/rerender/Rerender.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState, type ReactNode } from 'react';
+
+interface RerenderProps {
+    /**
+     * The duration in milliseconds between each rerender.
+     * @default 1000
+     */
+    intervalDuration?: number;
+    /**
+     * Time-sensitive content to render.
+     */
+    children: (currentTime: number) => ReactNode;
+}
+
+/**
+ * Rerender component
+ *
+ * This component triggers a rerender at a specified interval, providing the current time
+ * (in milliseconds) to its child function. Useful for dynamically updating content based
+ * on time-sensitive data.
+ */
+export const Rerender = ({ intervalDuration = 1000, children }: RerenderProps): JSX.Element => {
+    const [time, setTime] = useState(Date.now());
+
+    useEffect(() => {
+        const interval = setInterval(() => setTime(Date.now()), intervalDuration);
+        return () => clearInterval(interval);
+    }, [setTime, intervalDuration]);
+
+    return <>{children(time)}</>;
```

Comment:
This can simply be:
```suggestion
    return children(time);
```

Reply:
Getting an error when no wrapping:
```
Type 'ReactNode' is not assignable to type 'Element'.
  Type 'undefined' is not assignable to type 'ReactElement<any, any>'.
```

Reply:
That's because of the return type of the component set as `JSX.Element`

Reply:
✅